plugins/secrets/create.js

```diff
@@ -45,7 +45,19 @@ module.exports = () => ({
                                 `User ${username} is not an admin of this repo`);
                         }
                     })
-                    .then(() => secretFactory.create(request.payload))
+                    // check if secret already exists
+                    .then(() => secretFactory.get({
+                        pipelineId: request.payload.pipelineId,
+                        name: request.payload.name
+                    }))
+                    // if secret already exists, reject
+                    .then((secret) => {
+                        if (secret) {
+                            throw boom.conflict(`Secret already exists: ${secret.id}`);
```

Comment:
To make it more clear, can we put something like `Secret already exists with the id: ${secret.id}` ?